torchbenchmark/score/compute_score.py

```diff
@@ -8,31 +8,147 @@
+        name, mean = b['name'], b['stats']['mean']
+        config, task, model_name = get_benchmark_configuration(name)
+        if not found_benchmarks[task]:
+            found_benchmarks[task] = defaultdict(list)
+        # Append the tuple(mean, config, model_name) for all the configs the
+        # benchmark was run with.
+        found_benchmarks[task][model_name].append((mean, config, name))
+    return found_benchmarks
+
+def get_score_per_config(data, b_weight, norm):
+    """
+    This function iterates over found benchmark dictionary
+    and calculates the weight_sum and benchmark_score.
+    A score_db is then constructed to calculate the cummulative
+    score per config. Here config refers to device, mode and test
+    configurations the benchmark was run on.
+
+    For eg., if the benchmark was run in eval mode on a GPU in Torchscript JIT,
+                config = (train, cuda, jit)
+
+    This helper returns the score_db .
+
+    """
+    found_benchmarks = construct_benchmark_run_db(data)
     weight_sum = 0.0
-    for name in config['benchmarks']:
-        cfg = config['benchmarks'][name]
-        weight, norm = cfg['weight'], cfg['norm']
-        weight_sum += weight
-        measured_mean = [b['stats']['mean'] for b in data['benchmarks'] if b['name'] == name]
-        assert len(measured_mean) == 1, f"Missing data for {name}, unable to compute score"
-        measured_mean = measured_mean[0]
-        if fake_data is not None and name in fake_data:
-            # used for sanity checks on the sensitivity of the score metric
-            measured_mean = fake_data[name]
-        benchmark_score = weight * math.log(norm / measured_mean)
-        # print(f"{name}: {benchmark_score}")
-        score += benchmark_score
+    score_db = defaultdict(float)
+
+    for task, models in found_benchmarks.items():
+        for name, all_configs in models.items():
+            weight = b_weight[task] * (1.0/len(all_configs))
+            for mean, config, benchmark in all_configs:
+                weight_sum += weight
+                benchmark_score = weight * math.log(norm[benchmark] / mean)
+                score_db[config] += benchmark_score
 
-    score = target * math.exp(score)
     assert abs(weight_sum - 1.0) < 1e-6, f"Bad configuration, weights don't sum to 1, but {weight_sum}"
-    return score
+    return score_db
 
+def compute_score(weight_db, data, target, norm):
+    """
+    This API calculates the total score for all the benchmarks
+    that was run  by reading the data (.json) file.
+    The weights are then calibrated to the target score.
+
+    # TODO:
+    # score_db: score_per_config for now is just calculated.
+    # This needs to passed to plot_sweep.py to plot all the
+    # scores per config.
+    """
+    score = 0.0
+    score_db = get_score_per_config(data, weight_db, norm)
+
+    for config, scores in score_db.items():
+        score += scores
+        score_db[config] = target * math.exp(scores * 0.125)
```

Comment:
this is confusing me, you're modifying score_db as you iterate through it?  but then you're not using score_db again, and you're returning score?